cms/admin/forms.py

```diff
@@ -166,7 +166,7 @@ class PageForm(PageAddForm):
     # moderation - message is a fake field
     moderator_message = forms.CharField(max_length=1000, widget=forms.HiddenInput, required=False)
     
-    redirect = forms.CharField(label=_('Redirect'), max_length=255, required=False,
+    redirect = forms.URLField(label=_('Redirect'), max_length=255, required=False,
```

Comment:
The regex `r'^(https?://|/|../)\S+$'` matches our basic criteria for good URLs, and `urlparse.urlparse("http://exam]ple.com")` raises `ValueError: Invalid IPv6 URL`, which fixes the other problem raise in the ticket.